corn-maze.lic

```diff
@@ -0,0 +1,632 @@
+            waitrt?
+          end
+        end
+      end
+      custom_move "#{x}"
+      Flags.reset('scarecrow_done')
+    end
+    finish_task
+  end
+
+  def mice_task
+    @mice_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        fput 'wave'
+        pause 1
+        waitrt?
+        DRC.fix_standing
+        fput "TASK"
+        custom_move "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def scream_task
+    @scream_path.each do |x|
+      exit if Flags['maze_done']
+      if x == 'scream'
+        wait_on_done_move
+        fput "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def weeds_task
+    @maze_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        fput 'pull weeds'
+        pause 1
+        waitrt?
+        while !checkstanding
+          fput 'stand'
+        end
+        DRC.fix_standing
+        fput "TASK"
+        custom_move "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def corn_task
+    @maze_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        DRC.bput('forage corn', 'You forage around for a moment')
+        pause 1
+        waitrt?
+      fput('drop corn') if /corn/ =~ left_hand || /corn/ =~ right_hand
+        DRC.fix_standing
+        fput "TASK"
+        custom_move "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def tokens_task
+    @maze_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        fput 'search'
+        pause 1
+        waitrt?
+        DRC.fix_standing
+        fput "TASK"
+        custom_move "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def grasshopper_task
+    @mice_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        fput 'search'
+        pause 1
+        waitrt?
+        DRC.fix_standing
+        custom_move "#{x}"
+        fput "TASK"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    echo "TURN IN TASK THEN RETURN HERE!"
+    finish_task
+  end
+
+  def landmarks_task
+    @landmark_path.each do |x|
+      exit if Flags['maze_done']
+      if !Flags['Task_done']
+        if DRRoom.room_objs.count > 0
+          touch_room
+        end
+        custom_move "#{x}"
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    finish_task
+  end
+
+  def touch_room
+    targets = DRRoom.room_objs.grep(/rake|stones|hay|wood|basket|boulder|wheel|spiderweb|straw|statue|pail|fence|wagon|burrow|hut|scarecrow/)
```

Comment:
💡 This would be good to extract out to the top of the file into a named var (like the directions)